### ISSUE TITLE
Win MSI | Optional post-install auto launch

### ISSUE THREAD

**Issue by tmontes** [mu-editor/pup#145]
WANT:

* Optionally have the MSI installer with a "installtion complete, click here to launch the app" thingie.

WONDERING:

* What should the default be?
* Inclined for minimalistic and when opting in, the thingie shows up with a [x] checked checkbox that will launch the application on exiting the installer.

Clean, user-friendly, implementation probably depends on #14.

**Comment by ntoll** [comment]
So, if I understand you correctly, there's some sort of post-install script run as part of the installer on Windows, which we could use to do all the `venv` / packaged wheels stuff..? This would be totally awesome... or have I misunderstood what this could potentially be..?

**Comment by tmontes** [comment]
(after having clarified @ntoll's note in a previous meeting)

SELF-THOUGHT:

* Let's try to address this sooner, rather than later.
* Moving to NEXT.

**Comment by tmontes** [comment]
SELF-NOTE:

* Ran a quick experiment with no (partial?) success: got the GUI to show a "launch after install" but launching did not go through. Did not investigate further.

POTENTIALLY USEFUL RESOURCES for further investigation:

* https://wixtoolset.org/documentation/manual/v3/howtos/ui_and_localization/run_program_after_install.html
* https://stackoverflow.com/questions/31305483/switches-in-wixshellexectarget
* http://windows-installer-xml-wix-toolset.687559.n2.nabble.com/Passing-command-line-arguments-to-an-app-launched-after-setup-td1366362.html
* http://windows-installer-xml-wix-toolset.687559.n2.nabble.com/Running-command-line-at-the-end-of-installation-td2269109.html
* https://blogs.blackmarble.co.uk/rfennell/2013/02/25/running-an-external-command-line-tool-as-part-of-a-wix-install/

**Comment by tmontes** [comment]
PROGRESS NOTE:

* The recipe in the first link above failed (that was my first attempt).
* Changing the `<CustomAction>` to the one in the second linked resource made it work:

```
        <Property Id="WIXUI_EXITDIALOGOPTIONALCHECKBOXTEXT" Value="Launch {{ cookiecutter.app_name }}" />
        <Property Id="WixShellExecTarget" Value="[#{{ cookiecutter.app_name }}.vbs]" />
        <CustomAction Id="LaunchApplication"
                      Impersonate="yes"
                      FileKey="filC896A503FA45D448D320D55E417CFB95"
                      ExeCommand="-m {{ cookiecutter.launch_module }}"
                      Return="asyncNoWait"
                      />
```

Unfortunatelly, getting the `pythonw.exe` file ID did not work "magically" with `[#pythonw.exe]`... In my successul attempt, I hard coded the value onto the `.wxs` file.

NEXT:

* Figure out a nice way of getting the necessary file ID to properly generate the WiX input file.

**Comment by tmontes** [comment]
CLOSING:

* Albeit somewhat limited, the just merged PR #148 serves Mu Editor's use case very nicely.
  (would be nice to support auto-launch without license and license without auto-launch -- future work?)
